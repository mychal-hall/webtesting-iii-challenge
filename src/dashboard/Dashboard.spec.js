import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/react/cleanup-after-each";
import renderer from "react-test-renderer";

import Dashboard from "./Dashboard";

afterEach(cleanup);

describe("<Dashboard />", () => {
  it("Matches snap", () => {
    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders", () => {
    render(<Dashboard />);
  });

  it("Renders all children", () => {
    render(<Dashboard />);

    expect(document.querySelector(".controls")).toBeTruthy();
    expect(document.querySelector(".display")).toBeTruthy();
  });
});
