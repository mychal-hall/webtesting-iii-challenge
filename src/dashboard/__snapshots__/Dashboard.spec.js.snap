// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Matches snap 1`] = `
<div
  class="display panel"
>
  <div
    class="led green-led"
  >
    Unlocked
  </div>
  <div
    class="led green-led"
  >
    Open
  </div>
</div>
`;
